pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -66,8 +77,119 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
-	c.Logger.Infof("original: %v", original)
+	// Don't modify the informer's copy
+	el := original.DeepCopy()
+
+	// Propagate labels from EventListener to Deployment
+	labels := make(map[string]string, len(el.ObjectMeta.Labels)+1)
+	for key, val := range el.ObjectMeta.Labels {
+		labels[key] = val
+	}
+	labels["app"] = el.Name
+
+	// Create the EventListener Deployment
+	c.Logger.Infof("creating EventListener %s deployment in namespace %s", el.Name, el.Namespace)
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			// Create the EventListener's Deployment in the same Namespace as where
+			// the EventListener was created
+			Namespace: el.Namespace,
+			// Give the Deployment the same name as the EventListener
+			Name: el.Name,
+			// If our EventListener is deleted, then its Deployment should be as well
+			OwnerReferences: []metav1.OwnerReference{*el.GetOwnerReference()},
+			Labels:          labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "event-listener",
+							Image: *elImage,
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: int32(8082),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	oldDeployment, err := c.KubeClientSet.AppsV1().Deployments(el.Namespace).Get(el.Name, metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			c.Logger.Errorf("Error getting Deployments: %s", err)
+			return err
+		}
+
+		// Create the EventListener Deployment
+		_, err = c.KubeClientSet.AppsV1().Deployments(el.Namespace).Create(deployment)
+		if err != nil {
+			c.Logger.Errorf("Error creating EventListener Deployment: %s", err)
+			return err
+		}
+	} else if !reflect.DeepEqual(oldDeployment, deployment) {
+		// Update the EventListener Deployment
+		_, err = c.KubeClientSet.AppsV1().Deployments(el.Namespace).Update(deployment)
+		if err != nil {
+			c.Logger.Errorf("Error updating EventListener Deployment: %s", err)
+			return err
+		}
+	}
+
+	// TODO: This is an example Service that we will probably want to modify in the future
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			// Create the EventListener's Service in the same Namespace as where the
+			// EventListener was created
+			Namespace: el.Namespace,
+			// Give the Service the same name as the EventListener
+			Name: el.Name,
+			// If our EventListener is deleted, then its Service should be as well
+			OwnerReferences: []metav1.OwnerReference{*el.GetOwnerReference()},
+			Labels:          labels,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{"app": el.Name},
+			Type:     corev1.ServiceTypeLoadBalancer,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: corev1.ProtocolTCP,
+					Port:     int32(8082),
```

Comment:
looks like 8082 is repeated a couple times, would be good to use a constant (or command line param? or even config map?) for it

Reply:
I think ideally this will eventually be provided in the EventListener spec (maybe we could even have the option to provide an entire ServiceSpec)